src/Propel/Runtime/Util/PropelDateTime.php

```diff
@@ -70,34 +71,53 @@ protected static function isTimestamp($value): bool
      * Usually `new \Datetime()` does not contain milliseconds so you need a method like this.
      *
      * @param string|null $time Optional, in seconds. Floating point allowed.
+     * @param string $dateTimeClass Optional, class name of the created object.
      *
      * @throws \InvalidArgumentException
      *
-     * @return \DateTime
+     * @return \DateTimeInterface An instance of $dateTimeClass
      */
-    public static function createHighPrecision(?string $time = null): DateTime
+    public static function createHighPrecision(?string $time = null, string $dateTimeClass = 'DateTime'): DateTimeInterface
```

Comment:
just to make sure that everything will work if someone delete DateTime from use statement we can set here '\DateTime' (not sure if DateTime::class will work though)

Reply:
Me neither, but I'd suggest to not change this in this PR as other methods in this class use this format too (a string).
Better to change this in a future commit separate from this change I think.